lib/src/commit.rs

```diff
@@ -128,4 +128,13 @@ impl Commit {
     pub fn committer(&self) -> &Signature {
         &self.data.committer
     }
+
+    /// A commit is discardable if it has one parent, no change from its
+    /// parent, and an empty description.
+    pub fn is_discardable(&self) -> bool {
+        if let [parent_commit] = &*self.parents() {
+            return self.tree_id() == parent_commit.tree_id() && self.description().is_empty();
+        }
+        false
```

Comment:
nit: It's cheaper to check the description first since that doesn't involve reading any other commits from the backend (even though I see that the old code didn't do that).

Reply:
Would you prefer to check the description even before using `self.parents()`?

Reply:
Yes, before looking up the commit object at least. So .parent_id() is fine to call. Actually, I don't think we call this more than once or twice per command, I think, so don't worry about it. Feel free to leave it as is.

Reply:
I've put it first.